orca/request_opendap.py

```diff
@@ -1,21 +1,30 @@
 import re
-from xarray import open_dataset
+from xarray import open_dataset, load_dataset
+from tempfile import NamedTemporaryFile
 
+from .split import split_url
 
-def request_opendap(url, out_file):
-    try:
-        d = open_dataset(url)
-        d.to_netcdf(out_file)
-    except Exception as e:
-        raise e
 
-    return out_file
+def request_opendap(url):
+    data = open_dataset(url)
+    size = data.nbytes
+
+    if size / 2 > 5e8:
+        urls = split_url(url, size)
+    else:
+        urls = [url]
+
+    output = []
+    for path in urls:
+        tmp = NamedTemporaryFile(suffix=".nc", dir="/tmp")
+        open_dataset(path).to_netcdf(tmp.name)
+        output.append(tmp)
+
+    return output
 
 
 def build_url(thredds_base, filepath, variable, lat, lon):
     variable_form = re.compile(r"([a-z]*)(\[\d*(:\d*){1,2}\])")
     var_name, var_time = variable_form.findall(variable)[0][:2]
```

Comment:
Might be a good idea to have a `try/except` here in case the matches (and the indices) don't match up.